java-client/src/main/java/org/opensearch/client/opensearch/core/search/Hit.java

```diff
@@ -107,7 +107,7 @@ public class Hit<TDocument> implements JsonpSerializable {
 
     private Hit(Builder<TDocument> builder) {
 
-        this.index = ApiTypeHelper.requireNonNull(builder.index, this, "index");
+        this.index = builder.index;
```

Comment:
How do you ensure that this is only for a inner hit ? And outer hits should enforce ```requireNonNull```.

Reply:
I do think this is a good call out; The problem is that the the InnerHit is in a List of [HitsMetadata](https://github.com/opensearch-project/opensearch-java/blob/0334c78c88205eb3b1fece45796125cbe92f9134/java-client/src/main/java/org/opensearch/client/opensearch/core/search/HitsMetadata.java#L56), which is then in the [InnerHitsResult](https://github.com/opensearch-project/opensearch-java/blob/0334c78c88205eb3b1fece45796125cbe92f9134/java-client/src/main/java/org/opensearch/client/opensearch/core/search/InnerHitsResult.java#L52).

So unless there is a file made explicit to represent an InnerHit (which I think is a valid path), the other option would be to re-populate the Index and the Id to the InnerHits as well, even if they match the parent.